### PR TITLE
APIGOV-17614 - report metrics to CONDOR

### PR DESCRIPTION
**Pull request by jcollins-axway**

### CODE REVIEW

**Review by sbolosan — APPROVED** [review]

**Review by sbolosan — APPROVED** [review]

**Review by sbolosan — APPROVED** [review]

**Review by kflan-io — APPROVED** [review]

**Review by dfeldick — APPROVED** [review]

**Review by dfeldick — APPROVED** [review]

**Review by sbolosan — APPROVED** [review]

**Review by vivekschauhan — APPROVED** [review]

File: pkg/agent/agent.go [hunk 1]
```diff
@@ -378,11 +377,11 @@ func mergeResourceWithConfig() {
 
 	switch getAgentResourceType() {
 	case v1alpha1.DiscoveryAgentResourceName:
-		mergeDiscoveryAgentWithConfig(agent.cfg)
+		mergeDiscoveryAgentWithConfig(agent.cfg.(*config.CentralConfiguration))
```
Comment (tsjohns9) [inline pkg/agent/agent.go:380]: Can we check the type once before the switch statement?

File: pkg/traceability/traceability.go [hunk 2]
```diff
@@ -35,6 +36,23 @@ const (
 	traceabilityStr               = "traceability"
 )
 
+var traceabilityClients []*Client
+
+// GetClient - returns a random client from the clients array
```
Comment (tsjohns9) [inline pkg/traceability/traceability.go:41]: Why is there a slice of trace clients, and why would we want a random one?

Comment (jcollins-axway) [reply pkg/traceability/traceability.go:41]: We can have multiple connections to condor, this helps on high volume servers.  random so the metric events are not constantly trying to use the same one.

File: pkg/config/centralconfig.go [hunk 3]
```diff
@@ -425,9 +426,9 @@ func (c *CentralConfiguration) GetBuildDataPlaneType() string {
 }
 
 // CanPublishMetricEvent - Returns flag to indicate agent can publish metric events
-// func (c *CentralConfiguration) CanPublishMetricEvent() bool {
```
[the remark this answers is not in the document]
Comment (jcollins-axway) [reply pkg/config/centralconfig.go:428]: this was uncommented in my commits

File: pkg/agent/agent.go (hunk 1 above)
Comment (jcollins-axway) [reply pkg/agent/agent.go:380]: This is validated as central config prior to this usage.

**Pull request merged**